forem/forem
# Site Config: Get Started Section and Show Required Fields

## What type of PR is this? (check all applicable)


## Description

In [PR 8435](https://github.com/thepracticaldev/dev.to/pull/8435), we had introduced required vs optional fields to the site configuration layer. When something is required we show a banner. However, currently as the interface stands, it is not easy to see which are the required fields on the site configuration page. Hence **I have introduced the 'Required' label next to fields that are required.** (See screenshots)

However, with only the label, it's still difficult to find the required fields because they are hidden within the sections that are collapsed by default, so a user would need to open up each section to find the required fields. 
Hence, the next natural step for me was to pull out the required fields in **a 'Get Started' section**, which at some point in the future will probably be its' own onboarding workflow (something like Lisa's designs that she walked us through). 

The Get started section is expanded by default when all the mandatory configs are not completed however when they have been, it collapses. Thanks @citizen428 for this idea 😁. In addition, I've pulled out the descriptions and the placeholders in a hash to make it a bit more dynamic, because when we have a Get Started section we don't want to keep hardcoding the values but we want to simply use a loop to go through the mandatory configs, and hence we need the associated description and placeholder for the config.

The one thing to note for _this feature is that it's not the highest priority at the moment and so we just need something that is going to take us to that next step, and make it less confusing for admins and developers setting up._ 

In the future I will clean up some of the template into a partial as well as clean up descriptions and placeholders.

## Related Tickets & Documents

## QA Instructions, Screenshots, Recordings
Go to http://localhost:3000/internal/config where you will see the following changes:

![screencapture-localhost-3000-internal-config-2020-07-13-16_15_02](https://user-images.githubusercontent.com/2786819/87314847-0fa22300-c524-11ea-8c71-97c20a906811.png)

In addition, you should be able to update fields from either the Site Config or the Get Started section and the changes should reflect on both. You will see "Required" tags on the mandatory fields, and finally the Get Started section will be expanded or contracted by default  based on whether you have filled out all the mandatory configs. 🙃

## Added tests?

- [x] yes, I've amended them

## Added to documentation?


## [optional] Are there any post deployment tasks we need to perform?
None

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/3h4Bo5fja4ZgYS0WEg/giphy.gif)
